docs/language-server/protocol-language-server.md

```diff
@@ -3918,6 +3971,363 @@ standard input.
+associated with the current project, including any project-specific overrides.
+
+This does not include local libraries not defined explicitly in the project's
+edition, even if they can be resolved as per `prefer-local-libraries` setting.
+To get local libraries that are not directly referenced in the edition, use
+[`library/listLocal`](#librarylistlocal) instead.
+
+#### Parameters
+
+```typescript
+{
+  editionName?: String;
+}
+```
+
+#### Result
+
+```typescript
+{
+  availableLibraries: [LibraryEntry];
+}
+```
+
+#### Errors
+
+- [`EditionNotFoundError`](#editionnotfounderror) indicates that the requested
+  edition, or an edition referenced in one of its parents, could not be found.
+- [`FileSystemError`](#filesystemerror) to signal a generic, unrecoverable
+  file-system error.
+
+### `library/listLocal`
+
+Lists all local libraries available in the system.
+
+#### Parameters
+
+```typescript
+null;
+```
+
+#### Result
+
+```typescript
+{
+  localLibraries: [LibraryEntry];
+}
+```
+
+#### Errors
+
+- [`FileSystemError`](#filesystemerror) to signal a generic, unrecoverable
+  file-system error.
+
+### `library/create`
+
+Creates a new project package, but placed not in the projects directory, but in
+the local libraries directory, so that other projects (including the current
+one) that can resolve local libraries will be able to import it.
+
+The created library package inherits all edition settings from the current
+project.
+
+The endpoint just returns an empty message at success. Once this operation
+finishes, the IDE can add the import `import <namespace>.<name>` to the open
+file which will import the newly created library (the IDE must also ensure that
+`prefer-local-libraries` is set to `true` or otherwise, it must add a proper
+override to its own edition file to see this local library). Once the import is
+added, the library will be loaded and its content root will be sent in a
+[`file/rootAdded`](#filerootadded) notification.
+
+#### Parameters
+
+```typescript
+{
+  namespace: String;
+  name: String;
+  authors: [String];
+  maintainers: [String];
+  license: String;
+}
+```
+
+#### Result
+
+```typescript
+null;
+```
+
+#### Errors
+
+- [`LibraryAlreadyExists`](#libraryalreadyexists) to signal that a library with
+  the given namespace and name already exists.
+- [`FileSystemError`](#filesystemerror) to signal a generic, unrecoverable
+  file-system error.
+
+### `library/publish`
+
+Publishes a library located in the local libraries directory to the main Enso
+library repository.
+
+If `bumpVersionAfterPublish` is set to true, after publishing the library, its
+version is bumped automatically, so that future publications will not clash
+versions. This is a temporary solution and in the longer-term it should be
+replaced with separate settings allowing to arbitrarily modify the library
+version from the IDE.
+
+#### Parameters
+
+```typescript
+{
+  namespace: String;
+  name: String;
+  authToken: String;
```

Comment:
Also now I'm thinking if this should be a single token or maybe a list of pairs `(headerName, headerValue)`? This way in the future maybe we could adapt this to publishing to custom repositories with different authentication.

But maybe we shouldn't overthink this now, as for custom repos we'll need to provide their URL anyway, we can just create a separate endpoint for publishing to custom repositories once it is needed, so maybe better to not overthink this right now.

Reply:
I think this can be modified later if necessary, let's keep it simple for now.